modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/validators.js

```diff
@@ -0,0 +1,366 @@
+	 * @param {!Object} validator Validator to check each item against.
+	 * @return {!function()}
+	 */
+	arrayOf(validator) {
+		if (isInvalid(validators.object(validator))) {
+			throwConfigError('object', validator, 'arrayOf');
+		}
+
+		return maybe((value, name, context) => {
+			const result = validators.array(value, name, context);
+			if (isInvalid(result)) {
+				return result;
+			}
+
+			return validateArrayItems(validator, value, name, context);
+		});
+	},
+	bool: buildTypeValidator('boolean'),
+	func: buildTypeValidator('function'),
+
+	/**
+	 * Creates a validator that checks for a value within a range.
+	 * @param {!Number} min The minimum value allowed.
+	 * @param {!Number} max The maximum value allowed.
+	 * @return {!function()}
+	 */
+	inRange(min, max) {
+		const minResult = validators.number(min);
+		const maxResult = validators.number(max);
+		if (isInvalid(minResult)) {
+			return minResult;
+		}
+		if (isInvalid(maxResult)) {
+			return maxResult;
+		}
+
+		return maybe((value) => {
+			const valueResult = validators.number(value);
+			if (isInvalid(valueResult)) {
+				return valueResult;
+			}
+
+			return value >= min && value <= max;
+		});
+	},
+
+	/**
+	 * Creates a validator that checks if a value is an instance of a given class.
+	 * @param {!function()} expectedClass Class to check value against.
+	 * @return {!function()}
+	 */
+	instanceOf(expectedClass) {
+		return maybe((value, name, context) => {
+			if (value instanceof expectedClass) {
+				return true;
+			}
+			const msg = `Expected instance of ${expectedClass}`;
+
+			return composeError(msg, name, context);
+		});
+	},
+	number: buildTypeValidator('number'),
+	object: buildTypeValidator('object'),
+
+	/**
+	 * Creates a validator that checks that the value it receives is an object,
+	 * and that all values within that object pass the given validator.
+	 * @param {!function()} validator Validator to check each object value against.
+	 * @return {!function()}
+	 */
+	objectOf(validator) {
+		if (isInvalid(validators.func(validator))) {
+			throwConfigError('function', validator, 'objectOf');
+		}
+
+		return maybe((value, name, context) => {
+			for (const key in value) {
+				if (isInvalid(validator(value[key]))) {
+					return composeError(ERROR_OBJECT_OF_TYPE, name, context);
+				}
+			}
+
+			return true;
+		});
+	},
+
+	/**
+	 * Creates a validator that checks if the received value matches one of the
+	 * given values.
+	 * @param {!Array} arrayOfValues Array of values to check equality against.
+	 * @return {!function()}
+	 */
+	oneOf(arrayOfValues) {
+		return maybe((value, name, context) => {
+			const result = validators.array(arrayOfValues, name, context);
+			if (isInvalid(result)) {
+				return result;
+			}
+
+			return arrayOfValues.indexOf(value) === -1
+				? composeError(
+						composeOneOfErrorMessage(arrayOfValues),
+						name,
+						context
+				  ) // eslint-disable-line
```

Comment:
What lint is being skipped here? It may not even be required.

Lint suppressions are usually a smell, but when they can't be avoided, the suppressions should always be as specific as possible anyway. ie. it should say `eslint-disable-line the-rule-name`. Otherwise this suppression may mask another issue (`other-rule-name`) because it is a blanket suppression.

Once we know what the suppression is for, we can consider ways to avoid the suppression.

I see some other suppressions further down, so same thing will apply to them.

Reply:
I sent https://github.com/liferay/liferay-frontend-projects/pull/392 to beef up our suppression guidelines a bit so that I don't have to repeat this each time I see it. 😁